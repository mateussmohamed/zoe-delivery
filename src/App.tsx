import React, { FC } from 'react'
import { hot } from 'react-hot-loader/root'
import { ApolloProvider } from '@apollo/react-hooks'
import { ThemeProvider } from 'styled-components'

import client from './apollo/client'
import Routes from './routes/Routes'
import GlobalStyle from './styles/global'
import theme from './styles/themes/default'

const App: FC = () => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      <ApolloProvider client={client}>
        <Routes />
      </ApolloProvider>
    </ThemeProvider>
  )
}

export default hot(App)
