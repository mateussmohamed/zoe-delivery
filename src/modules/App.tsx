import React, { FC } from 'react'
import { hot } from 'react-hot-loader/root'

interface Props {
  name: string
}

const App: FC<Props> = ({ name }) => <h1>{name}</h1>

export default hot(App)
