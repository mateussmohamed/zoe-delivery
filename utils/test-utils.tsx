import React, { FC } from 'react'
import { render, RenderResult, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import '@testing-library/jest-dom'

import theme from '../src/styles/themes/default'

console.error = jest.fn()

const AllTheProviders: FC = ({ children }) => {
  const history = createMemoryHistory()

  return (
    <Router history={history}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </Router>
  )
}

const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
