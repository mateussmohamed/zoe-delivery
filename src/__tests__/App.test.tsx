import React from 'react'

import { render } from 'test-utils'

import App from '../App'

describe('<App />', () => {
  test('renders a Root component', async () => {
    const { container } = render(<App />)

    expect(container).toMatchSnapshot()
  })
})
