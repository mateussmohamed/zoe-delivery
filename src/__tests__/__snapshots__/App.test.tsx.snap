// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders a Root component 1`] = `
<div>
  <div>
    <div
      class="sc-AxjAm ijigQT"
      data-testid="header"
    />
    <h1>
      Home
    </h1>
    <input
      type="search"
    />
    <a
      href="/products"
    >
      Products
    </a>
  </div>
</div>
`;
